### PR TITLE
Add conda recipe format v2 stdlib lint support

### PR DESCRIPTION
**Pull request by wolfv**
~Note: the addition of `recipe_dir` in the lint_stdlib function breaks API.~

No API breaks here.

### CODE REVIEW

**Review by beckermr — CHANGES_REQUESTED** [review]
We should pass the variant.yaml path as the conda_build_config_filename. This won't break the API and is consistent with how the function worked previously.

**Comment by wolfv** [comment]
Good point. I'll refactor it later and update the PR :)

**Comment by wolfv** [comment]
I removed the API breakage. I think this one should be good as well now.

File: tests/test_lint_recipe.py [hunk 1]
```diff
@@ -329,6 +358,144 @@ def test_license_file_empty(is_rattler_build):
...
+        (["10.9", "11.0"], ["10.13", "11.0"], ["10.12"], "You are"),
+        # only sdk -> no warning
+        (None, None, ["10.13"], None),
+        (None, None, ["10.14", "12.0"], None),
+        # only sdk, but below global baseline -> warning
+        (None, None, ["10.12"], "You are"),
+        (None, None, ["10.12", "11.0"], "You are"),
+    ],
+)
+def test_rattler_cbc_osx_hints(
```
Comment (beckermr) [inline tests/test_lint_recipe.py:412]: textwrap.dedent can be used here to make things a bit more legible.

**Review by beckermr — CHANGES_REQUESTED** [review]

File: tests/test_lint_recipe.py (hunk 1 above)
Comment (wolfv) [reply tests/test_lint_recipe.py:412]: agreed. We just copied the test from a few lines above though :)

Comment (nichmor) [reply tests/test_lint_recipe.py:412]: applied dedent

Comment (wolfv) [reply tests/test_lint_recipe.py:412]: thanks @nichmor

Comment (beckermr) [reply tests/test_lint_recipe.py:412]: Ahhhh sorry for the random request then. IDK what we did for the other one. Thanks for fixing here. Much appreciated!

**Review by beckermr — APPROVED** [review]

**Pull request merged**